### ISSUE TITLE
Accounts table missing (?)

### ISSUE THREAD

**Issue by chivagio** [cioraneanu/firefly-pico#35]
Hi, I can't seem to make this work. I'm getting the following error message : 

```Error: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly-pico.accounts' doesn't exist (Connection: mysql, SQL: select * from `accounts` where `id` in (470, 683, 24, 78, 56, 257, 472, 618, 657, 663, 720, 361, 327, 340, 368, 271, 43, 567, 795, 393, 572, 796, 129, 207, 278, 561, 802, 665, 88, 741, 634, 702, 718, 610, 280, 722, 449, 710, 62, 184, 165, 715, 131, 54, 231, 302, 689, 643, 15, 381))```

My config : 
 ```
  firefly-pico:
    image: cioraneanu/firefly-pico:latest
    container_name: firefly_pico
    ports:
      - "6976:80"
    environment:
      - FIREFLY_URL=http://XXX.XXX.XXX.XXX:8081
      - DB_CONNECTION=mysql
      - DB_HOST=XXX.XXX.XXX.XXX
      - DB_PORT=3306
      - DB_DATABASE=XXXXX
      - DB_USERNAME=XXXXX
      - DB_PASSWORD=XXXXX
      - TZ=Europe/Brussels
```

I've checked the DB and I see 7 tables, but none called "accounts" : 
![image](https://github.com/cioraneanu/firefly-pico/assets/46843213/9a3fe309-963d-401c-9c1a-1e8ca6a130a6)

Am I doing something wrong ?

**Comment by cioraneanu** [comment]
You can try redeploying the container from scratch or ssh into the pico container and from `/var/www/html` run `php artisan migrate`

**Comment by chivagio** [comment]
Thanks for the quick response! 

I get this when running "php artisan migrate" from within the container : 
![image](https://github.com/cioraneanu/firefly-pico/assets/46843213/febe3ad3-4b1a-4dd9-b87f-60b301c96d54)

And in the Pico front same error message as above.

**Comment by cioraneanu** [comment]
From what I see it's a mysql DB related issue. [StackOverflow](https://stackoverflow.com/questions/8434518/mysql-foreign-key-constraint-is-incorrectly-formed-error). Personally I recommned Postgres :)

One workaround would be to ssh into the cointainer, open the file `database/migrations/2023_10_02_124513_create_transaction_templates.php` and comment the line that adds the failing foreign key [This one](https://github.com/cioraneanu/firefly-pico/blob/main/back/database/migrations/2023_10_02_124513_create_transaction_templates.php#L35).

Then run the `migrate` command again. 

Bonus points inspect the resulted tables to check if the 2 columns had indeed different types (`transaction_templates.id `and `transaction_template_tags.transaction_template_id`)

Create the Foreign Key manually in your desired DB app.

**Comment by chivagio** [comment]
Yes, that's it! :) I've had to do the same for `2024_02_22_131702_create_transaction_template_audio_names_table.php`. Now the DB is correctly populated.

I've checked the 2 columns and they are indeed different.

`transaction_templates.id` is `bigint(20) unsigned`
`transaction_template_tags.transaction_template_id` is `int(11).`

Anyways, I've been able to change the columns to the correct type and manually create the FKs and everything is working great now!! :)

Many thanks for your quick responses and ofc for this wonderful work that you've put into this!
